### PR TITLE
Replace pullspecs replacement provided by freshmaker

### PR DESCRIPTION
**Pull request by tkdchen**
* CLOUDBLD-3976

Signed-off-by: Chenxiong Qi <cqi@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [n/a] JSON/YAML configuration changes are updated in the relevant schema
- [n/a] Changes to metadata also update the documentation for the metadata
- [ ] Pull request has a link to an osbs-docs PR for user documentation updates
- [ ] New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/plugins/pre_pin_operator_digest.py [hunk 1]
```diff
@@ -132,6 +141,20 @@ def run_in_orchestrator(self):
         else:
             if pullspecs:
                 replacement_pullspecs = self._get_replacement_pullspecs(pullspecs)
+
+                if self.operator_csv_modifications_url is not None:
```
Comment (ben-alkov) [inline atomic_reactor/plugins/pre_pin_operator_digest.py:145]: @tkdchen;
I will end up needing similar code (ending with `modification_data = resp.json()`) to modify other metadata. May I ask you to make this reusable in some way?

**Comment by ben-alkov** [comment]
@tkdchen; Looks like you need to rebase to get the schema validation stuff. Once you do that, I think you'll notice that some of the tests in test_pin_operator_digests are failing due to missing required fields (at first glance seems to be mostly 'metadata')

File: atomic_reactor/plugins/pre_pin_operator_digest.py [hunk 2]
```diff
@@ -132,6 +141,22 @@ def run_in_orchestrator(self):
...
+                    session = get_retrying_requests_session()
+                    resp = session.get(self.operator_csv_modifications_url)
+                    resp.raise_for_status()
+                    modification_data = resp.json()
```
[the remark this answers is not in the document]
Comment (tkdchen) [reply atomic_reactor/plugins/pre_pin_operator_digest.py:149]: @mkosiarc the validation will be done by another subtask. This PR is just for the subtask to replace pullspecs.

[the remark this answers is not in the document]
Comment (mkosiarc) [reply atomic_reactor/plugins/pre_pin_operator_digest.py:149]: I know, I was just wondering in which component and at what time this gets done

**Review by mkosiarc — APPROVED** [review]

**Comment by MartinBasti** [comment]
Closing in favor of https://github.com/containerbuildsystem/atomic-reactor/pull/1586

**Pull request closed without merging**